libs/datamodel/connectors/mongodb-datamodel-connector/src/lib.rs

```diff
@@ -45,18 +45,6 @@ impl Connector for MongoDbDatamodelConnector {
         referential_integrity.allowed_referential_actions(BitFlags::empty())
     }
 
-    fn validate_field(&self, field: &dml::field::Field, errors: &mut Vec<ConnectorError>) {
-        // If the field is _not_ a native-type-annotated field and it has a `dbgenerated` default, we error.
-        if !matches!(field.field_type(), FieldType::Scalar(_, _, Some(_)))
-            && matches!(field.default_value().as_ref().map(|v| v.kind()), Some(DefaultKind::Expression(expr)) if expr.is_dbgenerated())
-        {
-            errors.push(ConnectorError::from_kind(ErrorKind::FieldValidationError {
-                field: field.name().to_owned(),
-                message: "MongoDB `@default(dbgenerated())` fields must have a native type annotation.".to_owned(),
-            }));
-        }
-    }
```

Comment:
This is potentially controversial. This is not tested, and the purpose isn't completely clear. We can of course reimplement if it turns out we _do_ wan to validate this.

Reply:
I think we'd want a connector method for defaults validation, but I can't reimplement a validation I don't know and that isn't tested.

Reply:
Ideally we'd validate exactly this. As it is, the migration will probably break at some point and it's not clear why we always forbid that. Btw we were trying to get rid of `dbgenerated()` without argument, and now we're relying on it by design on mongo. I'm not a fan at all.

Reply:
Test and new code in a68199c

Reply:
Yeah I don't like this either. I also don't like to not have a validation for this. We should aim for better design with the dbgenerated for sure.